casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -99,16 +99,20 @@ final class Estimator[F[_]: Sync: Log: Metrics: Span](
       latestMessages <- latestMessagesHashes.values.toStream
                          .traverse(hash => blockDag.lookup(hash))
                          .map(_.flatten)
-      result <- if (latestMessages.isEmpty) {
+      topBlockNumber <- blockDag.latestBlockNumber
+      filteredLM = latestMessages.filter(
+        _.blockNum > topBlockNumber - Estimator.latestMessageMaxDepth
+      )
+      result <- if (filteredLM.isEmpty) {
                  genesis.blockHash.pure[F]
                } else {
-                 latestMessages
-                   .foldM(latestMessages.head) {
-                     case (acc, latestMessage) =>
+                 filteredLM
+                   .foldM(filteredLM.head) {
+                     case (acc, filteredLM) =>
```

Comment:
Now one item has the same name as the collection. Existing name seems reasonable.

Reply:
Thanks, indeed